internal/abci/interfaces.go

```diff
@@ -82,3 +67,15 @@ type ConsensusParams interface {
 	// We may want these to be separate in the future.
 	VotingPeriod() int64
 }
+
+// DB is the interface for the main SQL database. All queries must be executed
+// from within a transaction. A DB can create read transactions or the special
+// two-phase outer write transaction.
+type DB interface {
+	sql.OuterTxMaker
+	sql.ReadTxMaker
+	sql.SnapshotTxMaker
+	sql.Executor
+	sql.TxMaker
+	AutoCommit(auto bool)
+}
```

Comment:
```go
type DB interface {
	sql.OuterTxMaker
	sql.ReadTxMaker
	sql.SnapshotTxMaker
}
```

is all that's required by `AbciApp`.

Having the other methods available are likely to lead to misuse.